src/main/java/com/ethanyidong/bunny/cmd/TokenizedCommand.java

```diff
@@ -0,0 +1,61 @@
+package com.ethanyidong.bunny.cmd;
+
+import java.util.HashMap;
+
+public class TokenizedCommand {
+    private final String command;
+    private String positionalArgument;
+    private HashMap<String, String> flagArguments;
+
+    public TokenizedCommand(String command) {
+        String[] commandAndArguments = command.split(" ", 2);
+        this.command = extractCommand(commandAndArguments[0]);
+
+        String arguments;
+        if (commandAndArguments.length == 2) {
+            arguments = commandAndArguments[1];
+        } else {
+            return;
+        }
+
+        String[] positionalAndFlagArguments = arguments.split("/");
+        this.positionalArgument = extractPositionalArgument(positionalAndFlagArguments);
+        this.flagArguments = extractFlagArguments(positionalAndFlagArguments);
+
+    }
+
+    private static String extractCommand(String command) {
+        return command;
+    }
+
+    private static String extractPositionalArgument(String[] positionalAndFlagArgument) {
+        return positionalAndFlagArgument[0].trim();
+    }
+
+    private static HashMap<String, String> extractFlagArguments(String[] positionalAndFlagArguments) {
+        HashMap<String, String> ret = new HashMap<>();
```

Comment:
Variable name `ret` may not be descriptive enough to explain what this hashmap does. Consider renaming it. [See here](https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#use-name-to-explain)